CMakeLists.txt doesn't respect CMAKE_INSTALL_PREFIX

```
cmake '-DCMAKE_INSTALL_PREFIX:PATH=/tmp' ..
...
make install
```

Actual result:
```
Install the project...
-- Install configuration: ""
-- Installing: /tmp/bin/fips
-- Installing: /var/lib/flatpak/exports/share/applications/fips.desktop
CMake Error at cmake_install.cmake:65 (file):
  file INSTALL cannot copy file "/home/matwey/lab/fips3/build/fips.desktop"
  to "/var/lib/flatpak/exports/share/applications/fips.desktop".
```

Expected result: files installed to prefixed locations

However, currently, we use cmake install with `-DCMAKE_INSTALL_PREFIX:PATH=/usr` in RPM building. This produces correct resuilt. The binary goes to `/usr/bin` and desktop goes to `/usr/share`

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -133,10 +133,7 @@ elseif(WIN32)
 	")
 	include(CPack)
 elseif(UNIX)
-	find_path(XDG_PREFIX
-		NAMES applications icons
-		PATHS /usr/share /usr/local/share
-		NO_DEFAULT_PATH)
+	set(XDG_PREFIX "${CMAKE_INSTALL_PREFIX}/share")
 	set(XDG_DESKTOP_DIR "${XDG_PREFIX}/applications")
 	set(XDG_ICONS_DIR "${XDG_PREFIX}/icons")
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -135,7 +135,6 @@ elseif(WIN32)
 elseif(UNIX)
 	find_path(XDG_PREFIX
 		NAMES applications icons
-		HINTS ENV XDG_DATA_DIRS
 		PATHS /usr/share /usr/local/share
 		NO_DEFAULT_PATH)
 	set(XDG_DESKTOP_DIR "${XDG_PREFIX}/applications")

